README.md

```diff
@@ -207,80 +207,139 @@ fi
 
-```sh
+```
 git config --global user.name
 git config --global user.email
 ```
 
 Now we can edit with and enter name in alphabetical order:
 
-```sh
+```
 code AUTHORS
 ```
 
 We need to do
 
-```sh
+```
 git add .
 ```
 
 We commit file with command
 
-```sh
+```
 git commit
 ```
 
 Next we execute the command:
 
-```sh
+```
 gh pr create
 ```
 
 Now we switch to the main branch
 
-```sh
+```
 git checkout main
 ```
 
 Delete old branch
 
-```sh
+```
 git branch -D name/add-to-author
 ```
 
 Finally, we run and wait for review.
 
-```sh
+```
 git pull
 ```
+=======
+## Main Quest: Moving from a JavaScript file to TypeScript
+
+our first step here is using a package manager(in our case Yarn) to install:
+```
+yarn add typescript
+```
+
+now we're going to rename our .js file to .ts using the following commands:
+
+```
+mv program_name.js program_name.ts
+```
+In the program we just renamed above we'll have to replace:
+``` const express = require('express') ``` for ```import express from 'express';```
+
+Lastly we're going to run this sequence of commands to finalize the setup:
+```
+yarn add @types/express
+yarn add ts-node
+```
+
+With those steps you should now be able to run your typescript code:
+```yarn ts-node program_name.ts```
+
+   congrats! __+100xp__
+   
+~ To learn more about the import and module lore in typescript refer to https://www.typescriptlang.org/docs/handbook/2/modules.html
+
+
+## Side Quest: Rebasing
+
+ Rebasing is the process of moving or combining a sequence of commits from one branch to another. It allows you to modify the commit history of a branch, making it appear as if you had made your changes directly on top of the latest commit in another branch.
+
+ Here's a visual example of  rebasing a Your_Branch to main:
+```
+       A---B---C  (Your_Branch)
+     /
+D---E---F---G   (main)
+
+
+              A---B---C  (Your_Branch)
+             /
+D---E---F---G  (main)
+```
+Assuming you are in Your_Branch, to perform what is ilustrated above execute the following:
+'''git rebase main'''
+
+To check if you should rebase you might want to use the command '''git log''' to view the commit history of your branch. You'll be able to see the latest commit on your branch.
+
+Compare the commit history of your branch with the target branch. If there are new commits on the target branch that you want to incorporate into your work, it's a good indication that you might need to perform a rebase.
+
+Additionally, remember that rebasing modifies the commit history, so if you have already pushed your branch to a remote repository, you will need to force push the rebased branch using the command '''git push -f'''. However, be cautious when force pushing, as it can cause issues if others are also working on the same branch.
+
+  bonus __+50xp__ and new skill acquired: __Forcefull Surge__(AoE spell) - Unleash the power of the -f flag to aggressively rewrite history, leaving a trail of chaos in your wake.
```

Comment:
is it too OP to give noobs such power right off the bat? could be game breaking xD @miorel  @BitCrusader

Reply:
Haha, some things have to be learned the hard way 🙂